app/lib/theme/widgets/ReceivedDateWidget.tsx

```diff
@@ -0,0 +1,163 @@
+} from "@fortawesome/free-solid-svg-icons";
+
+const ReceivedDateInput = forwardRef<HTMLDivElement, WidgetProps>(
+  ({ onClick, value, label, formContext }, ref) => {
+    let overdue;
+    const displayString = useMemo(() => {
+      const dueDate = DateTime.fromISO(formContext?.dueDate, {
+        setZone: true,
+        locale: "en-CA",
+      });
+      const formattedDueDate = dueDate.toFormat("MMM dd, yyyy");
+
+      const receivedDate = DateTime.fromISO(value, {
+        setZone: true,
+        locale: "en-CA",
+      });
+      const formattedReceivedDate = receivedDate.toFormat("MMM dd, yyyy");
+
+      if (!dueDate.day && !receivedDate.day) {
+        return "Select a date";
+      }
+
+      const currentDate = DateTime.now()
+        .setZone("America/Vancouver")
+        .endOf("day");
+
+      const dueDiff = Interval.fromDateTimes(currentDate, dueDate);
+      const daysFromToday = Math.floor(dueDiff.length("days"));
+
+      const lateDiff = Interval.fromDateTimes(dueDate, currentDate);
+      const lateDays = Math.floor(lateDiff.length("days"));
+
+      if (receivedDate.day) {
+        return (
+          <>
+            <div className="receivedStringWrapper">
+              <span style={{ marginRight: "1em" }}>Received</span>
+              <FontAwesomeIcon icon={faCheck} color={"green"} />
+            </div>
+            ({formattedReceivedDate})
+          </>
+        );
+      }
+      if (!receivedDate.day && currentDate > dueDate) {
+        overdue = true;
+        return <span className="overdue">Overdue by {lateDays} days</span>;
+      }
+
+      return daysFromToday > 60
+        ? `Due in ${Math.floor(
+            dueDiff.length("weeks")
+          )} weeks (${formattedDueDate})`
+        : `Due in ${daysFromToday} ${
+            daysFromToday === 1 ? "day" : "days"
+          } (${formattedDueDate})`;
+    }, [formContext?.dueDate, value]);
+
+    return (
+      <div className="receivedDateWrapper">
+        <div onClick={onClick} ref={ref} aria-label={label}>
+          {displayString}
+          <CalendarTodayIcon style={{ color: "black" }} />
+
+          <style jsx>{`
+            div {
+              display: flex;
+              align-items: center;
+              justify-content: space-between;
+              border: 2px solid #606060;
+              border-radius: 0.25em;
+              padding: 9px 9px 9px 9px;
+              font-size: 14.4px;
+              width: 300px;
+            }
+            :global(.receivedDateWrapper) {
+              display: flex;
+              align-items: center;
+              justify-content: flex-start;
+            }
+            :global(.receivedStringWrapper) {
+              display: flex;
+              align-items: center;
+            }
+
+            :global(.overdueIcon) {
+              margin-left: 1em;
+            }
+            div:hover {
+              cursor: pointer;
+            }
+            :global(.overdue) {
+              color: #cc0003;
```

Comment:
We have been using #cd2026 for error red in all other fields, which passes the accessibility checks.
```suggestion
            :global(.overdue) {
              color: #cd2026;
```